feat(analytics): GPU Usage

0xnirmal opened the pull request:
- Add Query for GPU Utilization (in aggregate) across all providers
- Add Query for GPU supply, leases, and utilization by vendor and model
- Add endpoint support on `v1/graph-data/` for GPU Utilization (aggregate) 

Todo: 
- Update UI to show GPU Utilization (aggregate) 
- ~Endpoint support for per vendor / model (will chat with Max on best approach here)~

Update:
- Added new endpoint support for vendor / model (if not provided will show all)

0xnirmal committed:
add queries + support GPU utilization in /graph-data

remove time comment

Fix import order (maybe)

add new route service vendor / model utilization

eslint fixes

verbiage

baktun14 commented on apps/api/src/services/db/gpuBreakdownService.ts:
```diff
@@ -0,0 +1,163 @@
+  );
+}
+
+export async function getLatestGpuBreakdown(): Promise<GpuBreakdownData[]> {
```
Can we expose this method as a documented route please?

0xnirmal replied:
On it - okay to take as a follow-up PR @baktun14?

baktun14 replied:
You can add it to this pr, you just need to add a route.

0xnirmal replied:
Done

ygrishajev commented on apps/api/src/services/db/gpuBreakdownService.ts:
```diff
@@ -0,0 +1,163 @@
+
+export async function getGpuUtilization() {
+  return await cacheResponse(
+    60 * 5, // 5 minutes
```
there are relevant constants in `date-fns`, the comment wouldn't be needed

0xnirmal replied:
Removed ty

0xnirmal commented:
Added Swagger API support. Lmk if any feedback on style - I added it to the new `GPU` section with optional params for vendor / model. If not provided, will return all.

https://github.com/user-attachments/assets/7bb49ab7-bb4d-4356-a349-22ea44987bb7

0xnirmal commented:
Ready for re-review @baktun14

baktun14 reviewed (dismissed):
Lgtm thanks for your contribution!

0xnirmal commented on apps/api/src/routes/v1/index.ts:
```diff
@@ -1,46 +1,47 @@
 import validators from "./validators/list";
 import versionMainnet from "./version/mainnet";
 import versionSandbox from "./version/sandbox";
 import versionTestnet from "./version/testnet";
```
Will fix import order, fixing commits first
